### PR TITLE
Add travis auto-release

### PR DESCRIPTION
**Pull request by abeltay**
I'll need to tag a release before I can test this on the public repo.

References:
https://docs.travis-ci.com/user/deployment/releases/
https://github.com/golang/dep/blob/master/.travis.yml

### CODE REVIEW

File: hack/build-all.bash [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# This script will build and calculate hash for each
```
Comment (soedar) [inline hack/build-all.bash]: why is the folder called `hack`? just curious.

Comment (abeltay) [reply hack/build-all.bash]: It can be called anything. It's hacked on top of the original repo. https://en.wikipedia.org/wiki/Kludge#Computer_science

Comment (soedar) [reply hack/build-all.bash]: Umm yeah I know what a hack is. If it technically is part of the release process... it isn't really a hack, is it?

Comment (abeltay) [reply hack/build-all.bash]: Sure, give me the name you want and I'll rename the file.

Comment (soedar) [reply hack/build-all.bash]: I'm just curious. The name doesn't matter

**Pull request merged**